src/frontend/src/binder/expr/function.rs

```diff
@@ -166,7 +169,54 @@ impl Binder {
                 return Ok(ExprImpl::literal_varchar(self.db_name.clone()));
             }
             "current_schema" if inputs.is_empty() => {
-                return Ok(ExprImpl::literal_varchar(DEFAULT_SCHEMA_NAME.to_string()));
+                return Ok(self
+                    .catalog
+                    .first_valid_schema(
+                        &self.db_name,
+                        &self.search_path,
+                        &self.auth_context.user_name,
+                    )
+                    .map(|schema| ExprImpl::literal_varchar(schema.name()))
+                    .unwrap_or_else(|_| ExprImpl::literal_null(DataType::Varchar)));
+            }
+            "current_schemas" => {
+                if inputs.len() != 1
+                    || (!inputs[0].is_null() && inputs[0].return_type() != DataType::Boolean)
+                {
+                    return Err(ErrorCode::ExprError(
+                        "No function matches the given name and argument types. You might need to add explicit type casts.".into()
+                    )
+                    .into());
+                }
+
+                let literal = try_match_expand!(&inputs[0], ExprImpl::Literal).unwrap();
```

Comment:
`unwrap` means we will panic if the user uses `current_schemas(true and false)`, but we should return an unsupported error instead:
```
let literal = try_match_expand!(&inputs[0], ExprImpl::Literal).map_err(|_| ErrorCode::NotImplemented)?;
```
It might be easier to use the let-else syntax:
```
let ExprImpl::Literal(literal) = &inputs[0] else { return Err(ErrorCode::NotImplemented); };
```